### PR TITLE
Jacobian allow list argnum

### LINKED ISSUE #112: Top-level Jacobian only works with integer argnum (not list)
This seems to be due to a bug in how we are interfacing with autograd, or -- less likely -- a bug in autograd itself.

If the user provides a quantum function `func` with multiple input arguments in its signature, we can only successfully call `qml.jacobian(func, argnum=0)`, `qml.jacobian(func, argnum=1)`, etc., but not `qml.jacobian(func, argnum=[0,1])` (raises some error inside autograd). 

Since we've coded `jacobian` to behave nicely if all the arguments are combined into an array, a suitable alternate usage is available. Would be nice to figure this bug out at some stage

### PR DESCRIPTION
**Pull request by cgogolin**
This fixes #112  and shouldn't have any adverse side-effects.

### CODE REVIEW

File: pennylane/__init__.py [hunk 1]
```diff
@@ -92,6 +92,7 @@
 """
 import os
 import logging as log
+import collections
```
Comment (co9olguy) [inline pennylane/__init__.py]: where is this import being used?

**Comment by josh146** [comment]
What is the status of this PR? Is it ready to merge, pending the removal of the `collections` import?

**Comment by cgogolin** [comment]
From my side it is ready to merge.

**Comment by quantshah** [comment]
If the tests pass, we could merge this. I resolved the conflicts and update to latest master.

**Comment by co9olguy** [comment]
Any reason not to merge this PR? It seems to have been neglected, but I'm not sure if there was a good reason to keep in on hold so long. @josh146 do you remember?

**Comment by josh146** [comment]
> Any reason not to merge this PR?

Not really. It was only not merged due to (a) your comment not being answered, and (b) not being very high priority (since no-one uses the jacobian function)

**Pull request merged**